pkg/api/vlabs/types.go

```diff
@@ -543,6 +543,7 @@ type AgentPoolProfile struct {
 	ProximityPlacementGroupID         string            `json:"proximityPlacementGroupID,omitempty"`
 	OSDiskCachingType                 string            `json:"osDiskCachingType,omitempty"`
 	DataDiskCachingType               string            `json:"dataDiskCachingType,omitempty"`
+	VMSSName                          string            `json:"vmssName,omitempty"`
```

Comment:
should this field be documented as READ-ONLY? From my understanding of the code above, any value set by the user will be ignored and get overwritten by the default.